### PR TITLE
register

### PR DESCRIPTION
**Pull request by savinmikhail**

### CODE REVIEW

File: internal/common/repository/register_repository.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	return nextId, nil
+}
+
+func (repo RegisterRepositoryPostgresql) CreateUser(login, password string) error {
+	id, err := repo.getNextId()
+
+	if err != nil {
+		return err
+	}
+	role := "ROLE_USER"
```
Comment (pmentoring) [inline internal/common/repository/register_repository.go]: magic inline constant

File: internal/common/service/login_service.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"golang.org/x/crypto/bcrypt"
+	"shortener-auth/internal/common/repository"
+)
+
+type LoginService struct {
+	repo       repository.LoginRepository
+	jwtService *JWTService
+}
+
+func (l LoginService) Login(login string, password string) (string, error) {
```
Comment (pmentoring) [inline internal/common/service/login_service.go]: login, password string

Comment (pmentoring) [inline internal/common/service/login_service.go]: l *LoginService

File: internal/routing/routes_api_v1.go [hunk 3]
```diff
@@ -8,5 +8,8 @@ import (
 )
 
 func Register(engine *gin.Engine, db *sql.DB, ctx *common.ApplicationContext) {
+	registerAction := appactions.NewRegisterAction(db, ctx)
```
Comment (pmentoring) [inline internal/routing/routes_api_v1.go]: thats not routing responsibility to create handlers

File: test/register_test.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	if err != nil {
+		return
+	}
+
+	ctx := common.NewApplicationContext("1", "", "trololo")
+	routing.Register(router, db, ctx)
+
+	w := httptest.NewRecorder()
+	request := http_actions.RegisterRequest{
+		Login:    "enisey",
```
Comment (pmentoring) [inline test/register_test.go:35]: typo in word BigDaddy

Comment (savinmikhail) [reply test/register_test.go:35]: ahahahah

File: internal/common/repository/login_repository.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
+package repository
+
+import (
+	"database/sql"
+	"shortener-auth/internal/common/model"
+)
+
+type LoginRepository interface {
```
Comment (pmentoring) [inline internal/common/repository/login_repository.go]: what is login repository? did we have login table in DB?

File: internal/common/service/jwt_service.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+	sub string
+	exp string
+	iat string
+}
+
+func NewJWTService(secret string) *JWTService {
+	return &JWTService{secretKey: secret}
+}
+
+func (s JWTService) CreateJwtForId(id int) (string, error) {
```
Comment (pmentoring) [inline internal/common/service/jwt_service.go:23]: for id of what? if its user id, name method properly

File: test/register_test.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	"net/http/httptest"
+	"shortener-auth/database"
+	"shortener-auth/internal/common"
+	"shortener-auth/internal/common/http_actions"
+	"shortener-auth/internal/routing"
+	"strings"
+	"testing"
+)
+
+func TestRegister(t *testing.T) {
```
Comment (pmentoring) [inline test/register_test.go]: where is 422, 401 tests?

Comment (savinmikhail) [reply test/register_test.go]: 401 on public route?

File: test/register_test.go [hunk 8]
```diff
@@ -54,7 +54,38 @@ func TestRegister(t *testing.T) {
...
 	require.True(t, token.Valid, "Token should be valid")
-	require.Same(t, token.Claims.GetIssuer(), "enisey")
+
+	claims, ok := token.Claims.(jwt.MapClaims)
+	require.True(t, ok, "Should be able to get claims")
+
+	assert.Greater(t, claims["sub"], float64(0), "Subject should match")
+}
+
+func TestRegisterBadRequest(t *testing.T) {
```
Comment (pmentoring) [inline test/register_test.go:67]: too much repeating code with happypass test

File: internal/common/http_actions/register.go [hunk 9]
```diff
@@ -15,10 +15,11 @@ type RegisterAction struct {
 }
 
 func NewRegisterAction(db *sql.DB, ctx *common.ApplicationContext) *RegisterAction {
+	repo := repository.NewUserRepository(db)
```
Comment (pmentoring) [inline internal/common/http_actions/register.go]: why http action creates repository?

File: internal/common/repository/user_repository.go [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+type UserRepositoryPostgresql struct {
+	conn *sql.DB
+}
+
+func NewUserRepository(conn *sql.DB) *UserRepositoryPostgresql {
+	return &UserRepositoryPostgresql{conn: conn}
+}
+
+func (r UserRepositoryPostgresql) getNextId() (int, error) {
```
Comment (pmentoring) [inline internal/common/repository/user_repository.go]: r *UserRepositoryPostgresql

File: internal/common/repository/user_repository.go [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+	err := result.Scan(&nextId)
+	if err != nil {
+		return 0, err
+	}
+
+	return nextId, nil
+}
+
+func (r UserRepositoryPostgresql) CreateUser(login, password string) error {
```
Comment (pmentoring) [inline internal/common/repository/user_repository.go]: r *UserRepositoryPostgresql

File: internal/common/repository/user_repository.go [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+	)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r UserRepositoryPostgresql) GetUserByLogin(login string) (*model.User, error) {
```
Comment (pmentoring) [inline internal/common/repository/user_repository.go]: r *UserRepositoryPostgresql

File: test/register_test.go (hunk 7 above)
Comment (pmentoring) [reply test/register_test.go]: im about tests at all. 401 on login can be

**Pull request merged**